engine/src/main/java/org/hibernate/search/backend/impl/lucene/MultiWriteDrainableLinkedList.java

```diff
@@ -37,17 +37,39 @@
 	 */
 	public void add(T element) {
 		final Node<T> newnode = new Node<T>( element );
-		addNode( newnode );
+		addNode( newnode, newnode );
 	}
 
-	private synchronized void addNode(Node<T> newnode) {
+	/**
+	 * Adds multiple new entries to this list, in the given order,
+	 * making sure no entry is added between the given elements.
+	 *
+	 * @param elements The elements to add.
+	 */
+	public void addAll(Iterable<T> elements) {
```

Comment:
This looks like very tricky, and there's a simpler approach ;)

See `org.hibernate.search.backend.impl.lucene.Changeset`